src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -617,6 +619,7 @@ public unsafe bool IsPrefix(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix
             return matched;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private unsafe bool StartsWithCore(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix, CompareOptions options, int* matchLengthPtr) =>
```

Comment:
`AggressiveInlining` on these small methods that are just a single `? :` and not getting naturally inlined due to too many arguments looks good. I would recommend submitting the change with just these `AggressiveInlining` annotations in a separate PR so that it does not get intermingled with the other changes proposed here.

Reply:
👍 I'm splitting the inlining changes out of this PR